cpp_code/tests/test_random_number_generator.cpp

```diff
@@ -249,43 +279,66 @@ static int random_seed(void) {
 }
 
 static void test_uniform_integer(RandomNumberGenerator &rng) {
-    vector<size_t> counts(PSI_DF);
     vector<double> probabilities(PSI_DF);
     size_t i;
+    unsigned passes, trials;
 
     for (i = 0; i < probabilities.size(); i++)
         probabilities[i] = 1/static_cast<double>(PSI_DF);
-    for (i = 0; i < NSAMPLES; i++)
-        counts[rng.nexti(PSI_DF)]++;
-    assert(psi_test(counts, probabilities, NSAMPLES));
+
+    for (passes = 0, trials = NTRIALS; trials --> 0;) {
```

Comment:
The arrow is cute, and I understand the intent, but it's likely to appear magical many readers. Please consider using the more obvious "trials > 0; --trials)"?